migration.rst

```diff
@@ -0,0 +1,414 @@
+        Request::setTrustedProxies(
+          explode(',', $trustedProxies),
+          Request::HEADER_X_FORWARDED_ALL ^ Request::HEADER_X_FORWARDED_HOST
+        );
+    }
+
+    if ($trustedHosts = $_SERVER['TRUSTED_HOSTS'] ?? $_ENV['TRUSTED_HOSTS'] ?? false) {
+        Request::setTrustedHosts([$trustedHosts]);
+    }
+
+    $kernel = new Kernel($_SERVER['APP_ENV'], (bool) $_SERVER['APP_DEBUG'], dirname(__DIR__));
+    $request = Request::createFromGlobals();
+    $response = $kernel->handle($request);
+
+    /*
+     * LegacyBridge will take care of figuring out whether to boot up the
+     * existing application or to send the Symfony response back to the client.
+     */
+    $scriptFile = LegacyBridge::prepareLegacyScript($request, $response, __DIR__);
+    if ($scriptFile !== null) {
+        require $scriptFile;
+    } else {
+        $response->send();
+    }
+    $kernel->terminate($request, $response);
+
+There are 2 major deviations from the original file. First of all we make
+``$kernel`` globally available. This allows us to use Symfony features inside
+our existing application and gives access to services configured in our
+Symfony application. This helps us prepare our own code to work better
+within the Symfony application when we transition it over or to replace
+outdated and redundant libraries with for example Symfony components.
+
+The legacy bridge, being called after the Symfony request is being handled,
+is responsible for figuring out which file should be loaded in order to
+process the old application logic. This can either be a front controller
+similar to Symfony's ``public/index.php`` or a specific script file based
+on the current route. The basic outline of this LegacyBridge could look
+somewhat like this::
+
+    namespace App;
+
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\Response;
+
+    class LegacyBridge
+    {
+        public static function prepareLegacyScript(Request $request, Response $response, string $publicDirectory): string
+        {
+            /*
+             * If Symfony successfully handled the route,
+             * we do not have to do anything.
+             */
+            if (false === $response->isNotFound()) {
+                return;
+            }
+
+            /*
+             * Figure out how to map to the needed script file
+             * from the existing application and possibly (re-)set
+             * some env vars.
+             */
+
+            return $legacyScriptFilename;
+        }
+    }
+
+This is the most generic approach you can take, that is likely to work
+no matter what your previous system was. You might have to account for
+certain "quirks", but since your original application is only started
+after Symfony finished handling the request you reduced the chances
+for side effects and any interference. Since the old script is called
+in the global namespace scope it will reduce side effects on the old
```

Comment:
```suggestion
in the global variable scope it will reduce side effects on the old
```

I would avoid the word "namespace" here because the reader might think you're referring to class namespaces.